src/Modules/ProjectModule/components/ProjectList/ProjectList.tsx

```diff
@@ -1,6 +1,162 @@
+import axios from "axios";
+import { SetStateAction, useContext, useEffect, useState } from "react";
+import DynamicHeader from "../../../SharedModules/components/DynamicHeader/DynamicHeader";
+import { AuthContext } from "../../../../context/AuthContext";
+import NoData from "../../../SharedModules/components/NoData/NoData";
+import { Button, Modal } from "react-bootstrap";
+import DeleteData from "../../../SharedModules/components/DeleteData/DeleteData";
+
+
 
 export default function ProjectList() {
+  const [ProjectsList, setProjectsList] = useState([]);
+  const [showIconIndex, setShowIconIndex] = useState(null);
+ let {baseUrl}= useContext(AuthContext)
+  const handleShowing = (index: any) => {
+    setShowIconIndex(index === showIconIndex ? null : index);
+  };
+  //  Delete Project Module 
+  const [showDelete, setDeleteShow] = useState(false);
+  const handleDeleteClose = () => setDeleteShow(false);
+  const handleDeleteShow = (id: SetStateAction<undefined>) => {
+    setItemId(id)
+    setDeleteShow(true);
+  }
+  const [itemId,setItemId] = useState();
+
+  // Get List
+  async function getProjectsList() {
+    try {
+      let response = await axios.get(
+        `${baseUrl}/Project/manager?title=&pageSize=&pageNumber=`,
+        {
+          headers: { Authorization: `Bearer ${localStorage.getItem("token")}` },
+        }
+      );
+      setProjectsList(response.data.data);
+      console.log(response.data.data, "FROM project");
+    } catch (error:any) {
+      console.log(error.data.data);
+
+    }
+  }
+
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+// Delete Project Api
+const deleteProject=async()=>{
+  try{
+const response= await axios.delete(`${baseUrl}/Project/${itemId}`,
+  {
+    headers:{ Authorization: `Bearer ${localStorage.getItem("token")}` },
+  }
+)
+console.log(response);
+handleDeleteClose();
+getProjectsList();
+
+  }catch(error){
+console.log(error);
+
+  }
+}
+
+
+  useEffect(() => {
+    getProjectsList();
```

Comment:
It is recommended by React to include anything declared outside of the effect inside the dependency array.  So here we should put the function inside the dependency array, but we will encounter another issue, the function is created in every rendered and this will cause the effect to run in every render and might cause an infinite loop.
We can solve this by two ways:
1- by using `useCallback` hook to preserve the reference of the function between renders.
2- make another state variable responsible for re-fethcing the data as we did yesterday.
![image](https://github.com/RanaFathy123/pms-project/assets/64327685/800a1af2-2293-43b9-91f2-3964b30a201d)
 
You can read more about this here in the React docs:
https://react.dev/reference/react/useEffect#specifying-reactive-dependencies
https://react.dev/reference/react/useEffect#removing-unnecessary-function-dependencies